src/python/bot/tasks/fuzz_task.py

```diff
@@ -1571,6 +1573,10 @@ def generate_blackbox_testcases(self, fuzzer, fuzzer_directory,
 
   def do_engine_fuzzing(self, engine_impl):
     """Run fuzzing engine."""
+    # For blackbox fuzzers, |FUZZ_TARGET| should be set to the fuzzer name.
+    if engine_impl.name == 'blackbox':
+      environment.set_value('FUZZ_TARGET', self.fuzzer_name)
+
     # Record fuzz target.
     fuzz_target_name = environment.get_value('FUZZ_TARGET')
     self.fuzz_target = record_fuzz_target(engine_impl.name, fuzz_target_name,
```

Comment:
This makes some of the rest of this function and a bit of the plumbing a lot messier. Any specific concerns here, or did you just want to avoid writing potentially confusing entries like blackbox_fuzzer_name that might show up in lists?

I worry that this might cause some issues down the road if we try to handle these as a special case, but if you have some specific concerns I can try to address them.

Reply:
Looking more, I'm less convinced we can actually get rid of this. It looks like we need the entity for functions like engine_common.get_all_issue_metadata_for_testcase.

Reply:
OK we can leave for now for testing if it's going to end up causing exceptions. However this doesn't really make sense to write so we should figure out a story for not doing this -- e.g. even for get_all_issue_metadata_for_testcase that's just the get the name of the binary so we can get TARGET.labels etc metadata files, which doesn't really apply here.

There might be more cases where things don't fit as-is or in a weird way, so maybe we should take a step back and see if there's some more general refactoring we can do to make things fit better rather than leaving things hackily like this.